pkg/operator/controller/extension/add.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package extension
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes/clientmap"
+	"github.com/gardener/gardener/pkg/operator/apis/config"
+	"github.com/gardener/gardener/pkg/operator/controller/extension/virtualcluster"
+)
+
+// AddToManager adds all Garden controllers to the given manager.
+func AddToManager(
```

Comment:
If there is only one controller and we don't plan for more, we don't need this `add.go` file at all

Reply:
I think that you will reasonably need something like the reference controller (for the infra resources in the runtime cluster) even if you decide to reconcile the all other aspects of the extensions in the same flow

Reply:
Can you explain?  We don't have such "reference controller" for the extension resources in seed clusters, do we? I didn't fully get what you mean.